src/vm/methodtable.cpp

```diff
@@ -6943,7 +6943,46 @@ MethodTable::FindDispatchImpl(
 }
 
 #ifndef DACCESS_COMPILE
-BOOL MethodTable::FindDefaultMethod(
+
+struct MatchCandidate
+{
+    MethodTable *pMT;
+    MethodDesc *pMD;
+};
+
+void ThrowExceptionForConflictingOverride(
+    MethodTable *pTargetClass,
+    MethodTable *pInterfaceMT,
+    MethodDesc *pInterfaceMD)
+{
+    LIMITED_METHOD_CONTRACT;
+
+    SString assemblyName;
+
+    pTargetClass->GetAssembly()->GetDisplayName(assemblyName);
+
+    SString strInterfaceName;
+    TypeString::AppendType(strInterfaceName, TypeHandle(pInterfaceMT));
+
+    SString strMethodName;
+    TypeString::AppendMethod(strMethodName, pInterfaceMD, pInterfaceMD->GetMethodInstantiation());
+
+    SString strTargetClassName;
+    TypeString::AppendType(strTargetClassName, pTargetClass);
+
+    COMPlusThrow(
+        kNotSupportedException,
+        IDS_CLASSLOAD_AMBIGUOUS_OVERRIDE,
+        strInterfaceName,
+        strMethodName,
+        strTargetClassName,
+        assemblyName);
+}
+
+// Find the default interface implementation method for interface dispatch
+// It is either the interface method with default interface method implementation, 
+// or an most specific interface with an explicit methodimpl overriding the method
+BOOL MethodTable::FindDefaultInterfaceImplementation(
```

Comment:
>FindDefaultInterfaceImplementation [](start = 18, length = 34)

This method is becoming increasingly complicated, with many levels of nested loops. Could you please refactor?